server/api.js

```diff
@@ -312,8 +313,50 @@ router.delete("/laptop_assignment/:assignmentId", async (request, response) => {
 				"(SELECT * FROM laptop_donation d WHERE d.id > $1 and (SELECT COUNT(*) FROM laptop_assignment a WHERE a.laptop_donation_id = d.id) < d.number_of_laptops ORDER BY d.id LIMIT 1)",
 				[queryResult.rows[0].laptop_donation_id]
 			);
+			const unAssignedRequests = await db.query(
+				"SELECT id, uuid, firstname FROM laptop_request  WHERE id > $1 and id NOT IN (SELECT laptop_request_id FROM laptop_assignment) and laptop_request_status != 'CANCELLED' ",
+				[queryResult.rows[0].laptop_request_id]
+			);
+
+			// assign next request to the rejected donation
+			if (
+				unAssignedRequests.rows != undefined &&
+				unAssignedRequests.rows.length > 0
+			) {
+				let getNumberOfLaptops = await db.query(
+					"SELECT number_of_laptops FROM laptop_donation WHERE id = $1",
```

Comment:
If you have a donation get unassigned, this only ever frees up 1 donation to be reassigned, so numberOfLaptops is wrong here

Reply:
Oh yes, that makes clears it up.

Reply:
I have implemented this and it still works as expected.